daemon/src/position_metrics.rs

```diff
@@ -382,6 +401,18 @@ mod metrics {
 
         set_metrics_for(POSITION_LONG_LABEL, status, &long);
         set_metrics_for(POSITION_SHORT_LABEL, status, &short);
+
+        let long_takers = long.iter().map(|cfd| cfd.counterparty_network_identity);
+        let short_takers = short.iter().map(|cfd| cfd.counterparty_network_identity);
+
+        let counterparties = long_takers.chain(short_takers).unique().count();
+
+        COUNTERPARTY_NUMBER_GAUGE
+            .with(&HashMap::from([
+                (POSITION_LABEL, POSITION_ALL_LABEL),
+                (STATUS_LABEL, status),
+            ]))
+            .set(counterparties as i64);
```

Comment:
🔧 With this snippet you're trying to capture the takers that have both long and short positions, right?

I think for that we need to do [this](https://programming-idioms.org/idiom/218/list-intersection/3834/rust).

Reply:
Yes, this should capture all takers which have either a long or a short position but don't count them double if they have both. 

The problem I was facing is that on line 405 we will get duplicated entries, hence, I have to call `unique`. 
So that I don't have to call `unique` twice, I merge the two vecs first (line 408) and then call `unique`.

Reply:
Hmmm, I'm not sure I understand it yet.

When we call `set_metrics_for` on lines 402 and 403 we're counting all takers with at least long and short positions respectively.

We then _also_ want to capture takers which have both long _and_ short positions. But there's gonna be an overlap between `long`, `short` and `all` the way this is arranged, right? Is that wanted?

Reply:
Yes, the thing is that `long+short!=all`, for example

`long=[taker1, taker,2]=2`
`short=[taker1, taker3]=2`
then we want all to be: 
`all=[taker1,taker2,taker3]=3`

Reply:
Okay, I misunderstood the meaning of `all` in this context.

It is not `all` in the sense of takers with all kinds of positions.

It is `all` in the sense of all the takers with any kind of position.

Thanks for clarifying and sorry for the confusion!

Reply:
You are welcome. Is there a better name for `all` which would make this easier to understand?

Reply:
Perhaps `any`? 

This value is for the `POSITION_LABEL` which I think indicates the type of position associated with a CFD. The values `long` and `short` are self-explanatory, but if we want to express that it could be either one `any` might be clearer.

It's almost like a wildcard or the absence of the label itself.